File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/QuizSkipMessage.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;

public record QuizSkipMessage(
        @Json(name = "string") //this is from the hack in Client.pollCommand
        String message
) implements Command {
    @Override
    public String commandName() {
        return CommandTypeOld.QUIZ_SKIP_MESSAGE.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/QuizSkipMessage.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.game

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType

@JvmRecord
@CommandType("quiz skip message")
data class QuizSkipMessage(
    @Json(name = "messageKey")
    val localisationKey: String,
    @Json(name = "messageData")
    val count: Count?
){
    @JvmRecord
    data class Count(
        @Json(name = "number")
        val votes: Int,
        @Json(name = "number2")
        val votesToPass: Int
    )
}